adapters/rhythmone/rhythmone.go

```diff
@@ -0,0 +1,148 @@
+package rhythmone
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/mxmCherry/openrtb"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/openrtb_ext"
+	"net/http"
+)
+
+type RhythmoneAdapter struct {
+	endPoint string
+}
+
+func (a *RhythmoneAdapter) MakeRequests(request *openrtb.BidRequest) ([]*adapters.RequestData, []error) {
+	errs := make([]error, 0, len(request.Imp))
+
+	var uri string
+	request, uri, errs = a.preProcess(request, errs)
+	if request != nil {
+		reqJSON, err := json.Marshal(request)
+		if err != nil {
+			errs = append(errs, err)
+			return nil, errs
+		}
+		if uri != "" {
+			headers := http.Header{}
+			headers.Add("Content-Type", "application/json;charset=utf-8")
+			headers.Add("Accept", "application/json")
+			return []*adapters.RequestData{{
+				Method:  "POST",
+				Uri:     uri,
+				Body:    reqJSON,
+				Headers: headers,
+			}}, errs
+		}
+	}
+	return nil, errs
+}
+
+func (a *RhythmoneAdapter) MakeBids(internalRequest *openrtb.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if response.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if response.StatusCode == http.StatusBadRequest {
+		return nil, []error{&errortypes.BadInput{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+
+	if response.StatusCode != http.StatusOK {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
+		}}
+	}
+	var bidResp openrtb.BidResponse
+	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+		return nil, []error{&errortypes.BadServerResponse{
+			Message: fmt.Sprintf("bad server response: %d. ", err),
+		}}
+	}
+
+	var errs []error
+	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+
+	for _, sb := range bidResp.SeatBid {
+		for i := range sb.Bid {
+			bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
+				Bid:     &sb.Bid[i],
+				BidType: getMediaTypeForImp(sb.Bid[i].ImpID, internalRequest.Imp),
+			})
+		}
+	}
+	return bidResponse, errs
+}
+
+func getMediaTypeForImp(impId string, imps []openrtb.Imp) openrtb_ext.BidType {
+	mediaType := openrtb_ext.BidTypeBanner
+	for _, imp := range imps {
+		if imp.ID == impId {
+			if imp.Banner != nil {
+				mediaType = openrtb_ext.BidTypeBanner
+			} else if imp.Video != nil {
+				mediaType = openrtb_ext.BidTypeVideo
+			}
+			return mediaType
+		}
+	}
+	return mediaType
+}
+
+func NewRhythmoneBidder(endpoint string) *RhythmoneAdapter {
+	return &RhythmoneAdapter{
+		endPoint: endpoint,
+	}
+}
+
+func (a *RhythmoneAdapter) preProcess(req *openrtb.BidRequest, errors []error) (*openrtb.BidRequest, string, []error) {
+	numRequests := len(req.Imp)
+
+	requestImpCopy := req.Imp
```

Comment:
Since `req.Imp` is a slice, and a slice is essentially a pointer to an existing array structure, `requestImpCopy := req.Imp` will end up referencing the same underlying array as `req.Imp` So it is not a true copy. 

You only seem to use `reqestImpCopy` for `imp := requestImpCopy[i]`, which will make `imp` a copy shallow of the original `imp`. So there does not seem to be a reason to define `reqestImpCopy`.